elastic/kibana
# [Ingest Pipelines] Error messages

## Summary

Compresses error feedback when attempting to create or update a pipeline.

Currently pipeline errors are reported as one continuous string (see screenshot). This PR adds code for parsing and displaying errors in a more user friendly way.

## How to test

1. Start Kibana on a basic license
2. Go the ingest pipelines app in the management section
3. Create a new pipeline, adding a processor with an empty `{}` for config (so you know it is invalid).
4. Duplicate the above processor a 6 times
5. Create the pipeline and the error messages should be present at the top of the creation form along with a toast of the error.

## Additional notes

Replicates the pattern in Snapshot & Restore of collapsible indices list

## Screenshots

**Current errors**

<img width="1184" alt="Screenshot 2020-06-18 at 11 59 00" src="https://user-images.githubusercontent.com/8155004/85987021-5ef93700-b9ed-11ea-906e-ac446803cdb7.png">

**After changes**

<img width="1449" alt="Screenshot 2020-06-30 at 11 53 33" src="https://user-images.githubusercontent.com/8155004/86112551-66414300-bac8-11ea-965b-27f13ff53141.png">

<img width="1262" alt="Screenshot 2020-06-30 at 11 53 53" src="https://user-images.githubusercontent.com/8155004/86112557-693c3380-bac8-11ea-879f-9f52d8fb425e.png">
